base/iterators.jl

```diff
@@ -152,15 +152,23 @@ end
 @inline done(r::Reverse{<:Enumerate}, state) = state[1] < 1
 
 """
-    Iterators.IndexValue(values, keys) <: AbstractDict{eltype(keys), eltype(values)}
+    Iterators.Pairs(values, keys) <: AbstractDict{eltype(keys), eltype(values)}
 
 Transforms an indexable container into an Dictionary-view of the same data.
 Modifying the key-space of the underlying data may invalidate this object.
 """
-struct IndexValue{K, V, I, A} <: AbstractDict{K, V}
+struct Pairs{K, V, I, A} <: AbstractDict{K, V}
     data::A
     itr::I
-    IndexValue(data::A, itr::I) where {A, I} = new{eltype(I), eltype(A), I, A}(data, itr)
+    Pairs(data::A, itr::I) where {A, I} = new{eltype(I), eltype(A), I, A}(data, itr)
+end
+
+function Base.showarg(io::IO, r::Pairs, toplevel)
+    if try typeof(pairs(r.data)) == typeof(r); catch; false; end
+        print(io, "pairs(", r.data, ")")
+    else
+        print(io, "Iterators.Pairs(",r.data, ", ", r.itr,")")
+    end
 end
```

Comment:
I think we usually put these definitions in show.jl to avoid the ordering issue. Also, missing a few spaces after the commas

Reply:
Can do.